pixelformat: Include compressed format in texture sets

thedmd opened the pull request:
This PR is follow up to comment: https://github.com/bkaradzic/bgfx/pull/2889#issuecomment-1224893315

This is how example present itself now with compressed format, this is first draft.
Some `texture_compression_*` textures do not really match. Maybe they can be split to different groups.
I'm open to suggestions on how to rearrange them.

Edit: Converted to draft to gather bit of the feedback before tidying up the changes.

![image](https://user-images.githubusercontent.com/1197433/186584213-2cf4f5c4-baf6-4a68-9cc0-6171de613889.png)

bkaradzic commented:
> This is how example present itself now with compressed format, this is first draft.

Looks good!

> Some texture_compression_* textures do not really match. Maybe they can be split to different groups.

What do you mean?

thedmd commented:
PTC12A, PTC24 look different than the other. Maybe other that GPU at hand does not support look different to, I didn't check that yet.

I can put them under different groups in combo box if necessary.

thedmd commented:
I added status for every texture. Format not supported by backend/GPU are colored red.
Status show why preview slot is not populated.
Also fixed off by one bugs. R1 is listed correctly (again), also BC1.

![image](https://user-images.githubusercontent.com/1197433/186606176-56a73256-32b2-4b7c-9854-5eafd646b361.png)

thedmd commented:
Preview for D3D9. Notice BC7 marked as not suppored.

![image](https://user-images.githubusercontent.com/1197433/186609453-d120db4e-10f0-4645-aa25-f270860ce5bf.png)

bkaradzic commented:
> I added status for every texture. Format not supported by backend/GPU are colored red.

Yes, that looks good!

> Status show why preview slot is not populated.

I already had this here: https://github.com/bkaradzic/bgfx/blob/30f2920ae5b27ca6c00c5263515b6fff9dbe723f/examples/47-pixelformats/pixelformats.cpp#L471-L477

bkaradzic commented:
> PTC12A, PTC24 look different than the other. Maybe other that GPU at hand does not support look different to, I didn't check that yet.

Ah, yeah that's bimg showing that there is no support for decompression...

> I can put them under different groups in combo box if necessary.

No. This is fine.
